test/chemistry/test_oovqe.py

```diff
@@ -0,0 +1,121 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test OOVQE """
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+from ddt import ddt
+from qiskit.aqua import aqua_globals
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.components.optimizers import COBYLA
+from qiskit.chemistry.core import Hamiltonian
+from qiskit.chemistry.drivers import PySCFDriver, UnitsType
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.components.initial_states import HartreeFock
+from qiskit.chemistry.core import TransformationType, QubitMappingType
+from qiskit.chemistry.algorithms.minimum_eigen_solvers import OOVQE
+from qiskit.aqua.operators.expectations import MatrixExpectation
+from qiskit import Aer
+
+
+@ddt
+class TestOOVQE(QiskitAquaTestCase):
+    """ Test of the OOVQE algorithm"""
+
+    def setUp(self):
+        super().setUp()
+        self.energy_vqe = -2.77  # energy of the VQE with pUCCD ansatz and LBFGSB optimizer
+        self.initial_point = [0.039374, -0.47225463, -0.61891996, 0.02598386, 0.79045546,
+                              -0.04134567, 0.04944946, -0.02971617, -0.00374005, 0.77542149]
+        self.seed = 50
+        aqua_globals.random_seed = self.seed
+        driver = PySCFDriver(
+            atom="H 1.738000 .0 .0; H .15148 1.73139 .0; H -1.738 .0 .0; H -0.15148 -1.73139 .0",
+            unit=UnitsType.ANGSTROM,
+            charge=0,
+            spin=0,
+            basis='sto3g')
+        self.qmolecule = driver.run()
+        self.core = Hamiltonian(transformation=TransformationType.FULL,
+                                qubit_mapping=QubitMappingType.PARITY,
+                                two_qubit_reduction=False,
+                                freeze_core=False,
+                                orbital_reduction=[])
+
+        algo_input = self.core.run(self.qmolecule)
+        self.qubit_op = algo_input[0]
+
+        init_state = HartreeFock(
+            num_orbitals=self.core._molecule_info['num_orbitals'],
+            qubit_mapping=self.core._qubit_mapping,
+            two_qubit_reduction=self.core._two_qubit_reduction,
+            num_particles=self.core._molecule_info['num_particles'])
+
+        self.var_form = UCCSD(
+            num_orbitals=self.core._molecule_info['num_orbitals'],
+            num_particles=self.core._molecule_info['num_particles'],
+            active_occupied=None, active_unoccupied=None,
+            initial_state=init_state,
+            qubit_mapping=self.core._qubit_mapping,
+            two_qubit_reduction=self.core._two_qubit_reduction,
+            num_time_slices=1,
+            method_doubles='pucc',
+            same_spin_doubles=False,
+            method_singles='both',
+            skip_commute_test=True,
+            excitation_type='d')
+
+        self.quantum_instance = QuantumInstance(Aer.get_backend('statevector_simulator'),
```

Comment:
Can we use BasicAer here which is always available. Otherwise it needs to check and skip test if Aer is not installed. I think this should be fine with BasicAer which we use for the majority of the tests.

Reply:
Yes, it works fine with it. Done.

Reply:
Great, thx. One thing with accessing the file can you change it so its done like here https://github.com/Qiskit/qiskit-aqua/blob/38f727b618b0b09a2aaf6deee8ad4e5fda7fb814/test/chemistry/test_driver_hdf5.py#L28 which uses a method that will compute the resource path relative to the test file. We have had problems with people running the tests in the past as to what their current folder is - by accessing it as above then as long as the test can be found/run its resource will be found always too.

Reply:
Great, I thought that might cause a problem. We will put it thanks.

Reply:
File names are now without path.